### PR TITLE
Product list component

### LINKED ISSUE #6: List Component
# User Story
As a **programmer**,
I want to **Implement a list component**,
so I can **Create a list which I want to use in further progress**.

# Description
image

#Material

# Done

- [x] Setup a basic input field
- [x] Display entered value below input field
- [x] Add hooks
- [x] Container for multiple lists
- [x] Create components
- [x] Create route for list item
- [x] Basic design

### LINKED ISSUE #12: List Item
# User Story
As a **programmer**,
I want to **have a product add function**,
so I can **create shopping list with items**.

# Description
image

#Material

# Done

- [x] Add  an example array of products
- [x] Create fetch api to products
- [x] Implement search function with display option
- [x] Add selected products to list component

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
Fix #12  

As a **programmer**,
I want to **have a product add function**,
so I can **create shopping list with items**.

 ### Note:
 - Design issues can be ignored. I know it's nice already
 - You can add your own written items to the list or the ones suggested. Latter will store a product Id in the database which will be used to compare in a later process. 

 ## Custom Routes to JSON server
https://www.npmjs.com/package/json-server#add-custom-routes

### CODE REVIEW

File: client/src/pages/List.js [hunk 1]
```diff
@@ -1,41 +1,97 @@
...
-        console.error(error);
-        setError(error);
-      }
-    }
-    fetchData();
-  }, [id]);
+  const { data: list, loading, error, refetch } = useAsync(fetchList, id);
+
+  async function handleClick(product) {
+    console.log(product);
```
Comment (mschmauder-hub) [inline client/src/pages/List.js:18]: Do you still need this console.log?

File: client/src/pages/List.js [hunk 2]
```diff
@@ -1,41 +1,97 @@
...
         <>
           <Container>
-            <h2>{list?.name}</h2>
+            <h2>{list.name}</h2>
             <div>
               <button onClick={() => history.goBack()}>Back</button>
             </div>
+
+            <Form onSubmit={handleSubmit}>
+              <label>Add your roducts</label>
```
Comment (mschmauder-hub) [inline client/src/pages/List.js]:
```suggestion
              <label>Add your products</label>
```

**Review by mschmauder-hub — COMMENTED** [review]
gj!

**Comment by LebkuchenCJ** [comment]
Adapt search bar, so you can send " " empty string with space

**Pull request merged**